arboretum/ibm_cloud/fetchers/fetch_cluster_resource.py

```diff
@@ -0,0 +1,165 @@
+            'org.ibm_cloud.cluster_resources.types', RESOURCE_TYPES_DEFAULT
+        )
+        cls.tempdir = tempfile.TemporaryDirectory()
+        return cls
+
+    @classmethod
+    def tearDownClass(cls):
+        """Cleanup class."""
+        cls.tempdir.cleanup()
+
+    def _get_iks_credentials(self, cluster, cluster_configs):
+        """Get credentials for an IKS cluster.
+
+        This function implements the procedure described in
+        https://cloud.ibm.com/apidocs/kubernetes#getclusterconfig
+        """
+        for name in cluster_configs[cluster['name']].namelist():
+            p = pathlib.Path(name)
+            if p.name.startswith('kube-config'):
+                kubeconfig = yaml.safe_load(
+                    cluster_configs[cluster['name']].read(name)
+                )
+                cluster_token = kubeconfig['users'][0]['user'][
+                    'auth-provider']['config']['id-token']
+            if p.name.endswith('.pem'):
+                cluster_configs[cluster['name']].extract(
+                    name, path=self.tempdir.name
+                )
+                ca_cert_filepath = os.path.join(self.tempdir.name, name)
+        return cluster_token, ca_cert_filepath
+
+    def _get_roks_credentials(self, cluster, api_key):
+        """Get credentials for a ROKS cluster.
+
+        This function implements the procedure described in
+        https://cloud.ibm.com/docs/openshift?topic=openshift-access_cluster#access_automation
+        """
+        s = self.session(cluster['serverURL'])
+        oauth_path = '/.well-known/oauth-authorization-server'
+        resp = s.get(oauth_path)
+        resp.raise_for_status()
+        token_endpoint = resp.json()['token_endpoint']
+        oauth_server = token_endpoint.split('/')[2]
+        s = self.session(f'https://{oauth_server}')
+        token_path = (
+            '/oauth/authorize?client_id='
+            'openshift-challenging-client&response_type=token'
+        )
+        resp = s.get(
+            token_path,
+            auth=('apikey', api_key),
+            headers={'X-CSRF-Token': 'a'},
+            allow_redirects=False
+        )
+        location = resp.headers['Location']
+        keyword = 'access_token='
+        start = location.find(keyword)
+        end = location.find('&', start)
+        cluster_token = location[start + len(keyword):end]
+        s.headers.update = ({'X-CSRF-Token': None})
+
+        return cluster_token, None
+
+    @store_raw_evidence('ibm_cloud/cluster_resources.json')
+    def fetch_cluster_resource(self):
+        """Fetch cluster resources."""
+        cluster_list_evidence = get_evidence_dependency(
+            'raw/ibm_cloud/cluster_list.json', self.locker
+        )
+        cluster_list = json.loads(cluster_list_evidence.content)
+        resources = {}
+        for account in cluster_list:
+            api_key = getattr(
+                self.config.creds['ibm_cloud'], f'{account}_api_key'
+            )
```

Comment:
Does this not work?:

```suggestion
            api_key = self.config.creds.get('ibm_cloud', f'{account}_api_key')
```